tools/get_vsdevcmd.ps1

```diff
@@ -0,0 +1,32 @@
+# Find all installed Visual Studio with Cmake, and only output the vsdevcmd path of the latest one
+PARAM([int]$outputEnv = 0)
+$latest_valid_vs = ""
+function choose_latter_vs {
+    param([string]$path)
+    if ($global:latest_valid_vs -lt $path) {
+        $cmake_path = $path + "\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe"
```

Comment:
Should we support checking for cmake.exe in the PATH by using Get-Command before requiring the VS install has one? If the user has setup their PATH to point to a specific cmake (which is what I've typically done), I would have expected we simply select the latest VS install. 

```
> get-command cmake

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     cmake.exe                                          3.25.2.0   C:\Program Files\CMake\bin\cmake.exe
```

Reply:
Now, most Python packages are tending to use cmake pip package and it was required by cibuildwheel, and it works pretty well for Linux an MacOS
However there are couple of some known cmake issues on some big projects like OpenCV for pip cmake on Windows. that's why there is some VS cmake looking up in this PR since that is most stable cmake variant for ci build.
And when you are building Python package, there already is a cmake.exe (installed by cibuildwheel), so I intentionally avoid search PATH for it. Given we only validate cmake VS in pipeline. Is it a good idea to enforce dev use the same one to avoid some cmake configuration issue?

Reply:
I think this is fine of a CI as we control the VS version installed.

Not sure I'd enforce it for dev usage though. Say someone only has VS 2019. I assume that means an older version of cmake. If they wanted to do some dev work with a 3rd party library that required a newer version the setup wouldn't support that by default. Maybe we could add an explicit env var for the cmake path to allow overriding.

Reply:
I just checked my local VS 2019 installation, which contains cmake 3.20. I guess VS 2019 have some service update to upgrade cmake from time to time.